infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -104,13 +114,47 @@
   }
 
   /**
-   * Creates specified pod.
+   * Starts the specified Pod via a Deployment.
    *
-   * @param pod pod to create
+   * @param pod pod to deploy
    * @return created pod
    * @throws InfrastructureException when any exception occurs
    */
   public Pod create(Pod pod) throws InfrastructureException {
+    putLabel(pod, CHE_WORKSPACE_ID_LABEL, workspaceId);
+    ObjectMeta metadata = pod.getMetadata();
+    PodSpec podSpec = pod.getSpec();
+    podSpec.setRestartPolicy("Always"); // Only allowable value
```

Comment:
Should we restart a pod if a container exits with the exit code 0? I'm not sure. It is kinda a situation that should not happen, right? So maybe it makes sense to use OnFailure? I don't have a strong opinion on that, just thinking about such a use case.

Reply:
Sadly, the only supported option in Deployments (at least on OpenShift) seems to be "Always". Setting it to "OnFailure" causes the `POST` request to be rejected. For what it's worth however, I manually failed a pod, and the workspace was detected as not running by the server (due to wsagent crashing) quickly and it showed a fairly understandable prompt to restart the workspace.

While the deployment does recreate a Pod that does nothing, it doesn't seem to have a significant negative impact. Hopefully in the future we'll be able to leverage the built in restart functionality to deal with failing workspaces through Kubernetes/OpenShift directly.